src/config.c

```diff
@@ -3198,6 +3198,7 @@ standardConfig static_configs[] = {
     createIntConfig("watchdog-period", NULL, MODIFIABLE_CONFIG | HIDDEN_CONFIG, 0, INT_MAX, server.watchdog_period, 0, INTEGER_CONFIG, NULL, updateWatchdogPeriod),
     createIntConfig("shutdown-timeout", NULL, MODIFIABLE_CONFIG, 0, INT_MAX, server.shutdown_timeout, 10, INTEGER_CONFIG, NULL, NULL),
     createIntConfig("repl-diskless-sync-max-replicas", NULL, MODIFIABLE_CONFIG, 0, INT_MAX, server.repl_diskless_sync_max_replicas, 0, INTEGER_CONFIG, NULL, NULL),
+    createIntConfig("hide-user-data-from-log", NULL, MODIFIABLE_CONFIG, -1, INT_MAX, server.hide_user_data_from_log, 0, INTEGER_CONFIG, NULL, NULL),
```

Comment:
```suggestion
    createBoolConfig("hide-user-data-from-log", NULL, MODIFIABLE_CONFIG, server.hide_user_data_from_log, 1, NULL, NULL),
```
I think you want this as a bool config.

Reply:
If it would be a bool config then we can't do https://github.com/valkey-io/valkey/pull/877#discussion_r1710211581
> Ideally we would also allow printing of other arguments that aren't keys, so I think this structure of a function makes more sense. I don't think we need to necessarily implement the logic to figure out which arguments are keys though.

Reply:
Oh, I think I wasn't clear about my suggestion. That suggestion was referring to the fact that we should only be omitting keys. Like if we do `SLOWLOG GET`, we will redact the `get` for no clear reason.

Reply:
Ok that makes more sense, so the index which is counted as sensitive data should be determained based on command type.

Reply:
> Ok that makes more sense, so the index which is counted as sensitive data should be determined based on command type.

Yes. I was saying that for now, we can just assume everything with index > 0 is possibly user down, and we can scope it down potentially in the future, since doing the getKeysFromCommand is a bunch of code for right now and I'm not sure it covers all cases.